packages/typespec-client-generator-core/test/package.test.ts

```diff
@@ -620,12 +622,17 @@ describe("typespec-client-generator-core: package", () => {
         }
       `);
       const sdkPackage = runner.context.experimental_sdkPackage;
-      strictEqual(sdkPackage.clients.length, 4);
+      strictEqual(sdkPackage.clients.length, 1);
 
-      const mainClient = sdkPackage.clients.find((c) => c.name === "TestServiceClient");
-      const fooClient = sdkPackage.clients.find((c) => c.name === "Foo");
-      const fooBarClient = sdkPackage.clients.filter((c) => c.name === "Bar")![0];
-      const barClient = sdkPackage.clients.filter((c) => c.name === "Bar")![1];
+      const mainClient = sdkPackage.clients[0];
+      const fooClient = mainClient.methods.find(
+        (m) => m.kind === "clientaccessor" && m.name === "getFoo"
+      )?.response as SdkClientType<SdkHttpOperation>;
+      const fooBarClient = fooClient.methods.find((m) => m.kind === "clientaccessor")
+        ?.response as SdkClientType<SdkHttpOperation>;
+      const barClient = mainClient.methods.find(
+        (m) => m.kind === "clientaccessor" && m.name === "getBar"
+      )?.response as SdkClientType<SdkHttpOperation>;
```

Comment:
the interface Bar inside namespace Foo is also called getBar? will it be problematic for naming conflicts with the client options?

Reply:
in previous reply, i think we need to find a way to show the path, so emitter could have an easy way to deal with this.